packages/ember-runtime/lib/system/core_object.js

```diff
@@ -418,7 +435,21 @@ CoreObject.PrototypeMixin = Mixin.create({
     @default false
     @public
   */
-  isDestroying: false,
+  isDestroying: descriptor({
+    get() {
+      let m = meta(this);
+      return (m._flags & SOURCE_DESTROYING) !== 0;
```

Comment:
Maybe I'm just old so it looks familiar but I don't find `if (m._flags & SOURCE_DESTROYING) {` unreadable or confusing.